clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -795,12 +795,30 @@ static void addSanitizers(const Triple &TargetTriple,
     PB.registerOptimizerLastEPCallback(SanitizersCallback);
   }
 
-  if (LowerAllowCheckPass::IsRequested()) {
+  // SanitizeSkipHotCutoffs: doubles with range [0, 1]
+  // Opts.cutoffs: ints with range [0, 1000000]
+  std::optional<std::vector<int>> scaledCutoffs =
+      CodeGenOpts.SanitizeSkipHotCutoffs.getAllScaled(1000000);
+
+  // TODO: remove IsRequested()
+  if (LowerAllowCheckPass::IsRequested() || scaledCutoffs.has_value()) {
     // We want to call it after inline, which is about OptimizerEarlyEPCallback.
     PB.registerOptimizerEarlyEPCallback([&](ModulePassManager &MPM,
                                             OptimizationLevel Level,
                                             ThinOrFullLTOPhase Phase) {
       LowerAllowCheckPass::Options Opts;
+
+      // TODO: after removing IsRequested(), the if case will be unconditional
```

Comment:
Opts.cutoffs = *scaledCutoffs;

Reply:
make vectors same type?

Reply:
unsigned makes sense for Opts.cutoffs. But it seems unnecessarily restrictive to force SanitizerMaskCutoffs::getAllScaled to return unsigned.

Reply:
we can relax that when needed
less code for now